src/NBitcoin/ScriptEvaluationContext.cs

```diff
@@ -1446,12 +1446,21 @@ private bool EvalScript(Script s, TransactionChecker checker, int hashversion)
                                             return SetError(ScriptError.InvalidStackOperation);
 
                                         int nKeysCount = new CScriptNum(this._stack.Top(-i), fRequireMinimal).getint();
-                                        if(nKeysCount < 0 || nKeysCount > 20)
-                                            return SetError(ScriptError.PubkeyCount);
 
-                                        nOpCount += nKeysCount;
-                                        if(nOpCount > 201)
-                                            return SetError(ScriptError.OpCount);
+                                        if (this.Network.Federations != null && lastOpcode?.Code == OpcodeType.OP_FEDERATION)
+                                        {
+                                            if (nKeysCount < 0 || nKeysCount > 150)
```

Comment:
150 signatures may have a significant impact. Perhaps a smaller number would be more sensible.